### PR TITLE
Visit by class name instead of key

### PR DESCRIPTION
**Pull request by edpaget**
Visit by class so we can be more flexible in defining workflows
Maybe revisit (heh) when the format is more locked down

This should fix the 500 errors @brian-c was seeing.

### CODE REVIEW

File: lib/tasks_visitor.rb [hunk 1]
```diff
@@ -15,17 +15,19 @@ def noop(n, p)
...
   end
 
-  def visit_hash(n, path)
+  def visit_Hash(n, path)
     n.each do |k, v|
       n[k] = visit(v, path + [k], k)
     end
   end
+  
+  alias :visit_ActiveSupport_HashWithIndifferentAccess :visit_Hash
```
Comment (camallen) [inline lib/tasks_visitor.rb]: Is this alias list exhaustive of the expected types? If not what about using `method_missing` to dispatch the relevant class method?

**Comment by brian-c** [comment]
I didn't realize the back end was actually reading anything in the workflow's tasks object. This is still a bit in flux since I've started work on the editor. Can we not care about what's in that until it's finalized?

**Comment by camallen** [comment]
@brian-c, we're reading the worfklow contents for create / update AND when serialising the workflow api responses. The original work was based on https://github.com/zooniverse/Panoptes/issues/212#issuecomment-59781884.

File: lib/tasks_visitor.rb (hunk 1 above)
Comment (edpaget) [inline lib/tasks_visitor.rb]: That should be an exhaustive list of types its possible to get back from
json.

**Comment by edpaget** [comment]
I want to make sure the translation stuff works right away. This code
should make sure that it didn't care what's in the workflow now (as opposed
to relying on the json object's key names as it was before) and just looks
for specific fields it knows to translate.

**Pull request merged**